server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -745,6 +748,108 @@ public DocTableInfo dropColumns(List<DropColumn> columns) {
         );
     }
 
+    private void validateRenameColumn(Reference refToRename, ColumnIdent newName) {
+        var oldName = refToRename.column();
+        var reference = getReference(oldName);
+        if (reference == null) {
+            reference = indexColumn(oldName);
+        }
+        if (!refToRename.equals(reference)) {
+            throw new ColumnUnknownException(oldName, ident);
+        }
+        if (getReference(newName) != null || indexColumn(newName) != null) {
+            throw new IllegalArgumentException("Cannot rename column to a name that is in use");
+        }
+    }
+
+    public DocTableInfo renameColumn(Reference refToRename, ColumnIdent newName) {
+        validateRenameColumn(refToRename, newName);
+        final ColumnIdent oldName = refToRename.column();
+
+        Predicate<ColumnIdent> toBeRenamed = c -> c.equals(oldName) || c.isChildOf(oldName);
+
+        // Renaming columns are done in 2 steps:
+        //      1) rename SimpleReferences' own ColumnIdents
+        //      2) rename dependencies such as GeneratedReferences, IndexReferences, Check Constraints, etc.
+        // where 1) is used to perform 2).
+
+        Map<ColumnIdent, Reference> oldNameToRenamedRefs = new HashMap<>();
+        for (var ref : references.values()) {
+            var r = toBeRenamed.test(ref.column()) ? ref.withReferenceIdent(
+                new ReferenceIdent(ref.ident().tableIdent(), ref.column().replacePath(newName))) : ref;
+            oldNameToRenamedRefs.put(
+                ref.column(), // old name
+                r); // renamed reference
+        }
```

Comment:
```suggestion
        for (var ref : references.values()) {
            ColumnIdent column = ref.column();
            if (toBeRenamed.test(column)) {
                var renamedRef = ref.withReferenceIdent(
                    new ReferenceIdent(ident, ref.column().replacePath(newName)));

                oldNameToRenamedRefs.put(column, renamedRef);
            }
        }
```

Looks like it's not necessary to add the references which aren't renamed. The lookups use `getOrDefault()`
And no need to add a comment with a better name if you can rename the variable itself.